### PR TITLE
zmqpb: add package

### PR DESCRIPTION
**Pull request by SFGrenade**

### CODE REVIEW

File: packages/z/zmqpb/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    on_fetch(function (package)
+        local result = {}
+        local libfiledir = (package:config("shared") and package:is_plat("windows", "mingw")) and "bin" or "lib"
+        result.links = "ZmqPb"
+        result.linkdirs = package:installdir("lib")
+        result.includedirs = package:installdir("include")
+        result.libfiles = path.join(package:installdir(libfiledir), "ZmqPb.lib")
+        return result
+    end)
```
Comment (waruqi) [inline packages/z/zmqpb/xmake.lua:22]: please add on_test

File: packages/z/zmqpb/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    set_homepage("https://github.com/SFGrenade/ZmqPb/")
+    set_description("A helper to use zeromq and protobuf together")
+    set_license("MPL-2.0")
+
+    set_urls("https://github.com/SFGrenade/ZmqPb/archive/refs/tags/v$(version).tar.gz",
+             "https://github.com/SFGrenade/ZmqPb.git")
+    add_versions("0.1", "4a34ec92faa381306356e84e2a2000093d8f76cfa037db1f4cd0adb0205faebb")
+
+    on_load(function (package)
+        package:set("installdir", path.join(os.scriptdir(), package:plat(), package:arch(), package:mode()))
```
Comment (waruqi) [inline packages/z/zmqpb/xmake.lua:11]: you should use on_install

File: packages/z/zmqpb/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    end)
+
+    on_fetch(function (package)
```
Comment (waruqi) [inline packages/z/zmqpb/xmake.lua:14]: please remove it.

File: packages/z/zmqpb/xmake.lua (hunk 2 above)
Comment (waruqi) [reply packages/z/zmqpb/xmake.lua:11]: refer https://github.com/xmake-io/xmake-repo/blob/81096aaf98fc85480a1e9aa341aaeee16e363373/packages/d/discord/xmake.lua#L28

Comment (waruqi) [reply packages/z/zmqpb/xmake.lua:11]: refer zlib, libspng package

File: packages/z/zmqpb/xmake.lua (hunk 1 above)
Comment (SFGrenade) [reply packages/z/zmqpb/xmake.lua:27]: i hope what i added for it works

File: packages/z/zmqpb/xmake.lua (hunk 2 above)
Comment (SFGrenade) [reply packages/z/zmqpb/xmake.lua]: added

File: packages/z/zmqpb/xmake.lua (hunk 3 above)
Comment (SFGrenade) [reply packages/z/zmqpb/xmake.lua]: removed

**Pull request merged**